Call sync_partition_metadata fails with No FileSystem

Steps to reproduce
1. Create partitioned hive table (you can skip if already exists)
2. Restart presto cluster
3. Call sync_partition_metadata function before executing any queries. 
>>>

1. Create partitioned table on hive
```sql
CREATE TABLE hive.default.page_views (
  view_time timestamp,
  user_id bigint,
  page_url varchar,
  ds date,
  country varchar
)
WITH (
  format = 'ORC',
  partitioned_by = ARRAY['ds', 'country'],
  bucketed_by = ARRAY['user_id'],
  bucket_count = 50
)
```

2. Restart presto cluster
3. Call sync_partition_metadata function
```
use hive.default;
call system.sync_partition_metadata('default', 'page_views', 'add');
```
```
io.prestosql.spi.PrestoException: No FileSystem for scheme: hdfs
    at io.prestosql.plugin.hive.SyncPartitionMetadataProcedure.syncPartitionMetadata(SyncPartitionMetadataProcedure.java:128)
    at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
    at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:649)
    at io.prestosql.execution.CallTask.execute(CallTask.java:160)
    at io.prestosql.execution.CallTask.execute(CallTask.java:60)
    at io.prestosql.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:168)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: No FileSystem for scheme: hdfs
    at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2658)
    at org.apache.hadoop.fs.PrestoFileSystemCache.createFileSystem(PrestoFileSystemCache.java:118)
    at org.apache.hadoop.fs.PrestoFileSystemCache.getInternal(PrestoFileSystemCache.java:90)
    at org.apache.hadoop.fs.PrestoFileSystemCache.get(PrestoFileSystemCache.java:63)
    at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:372)
    at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
    at io.prestosql.plugin.hive.HdfsEnvironment.lambda$getFileSystem$0(HdfsEnvironment.java:71)
    at io.prestosql.plugin.hive.authentication.UserGroupInformationUtils.lambda$executeActionInDoAs$0(UserGroupInformationUtils.java:29)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.security.auth.Subject.doAs(Subject.java:360)
    at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1742)
    at io.prestosql.plugin.hive.authentication.UserGroupInformationUtils.executeActionInDoAs(UserGroupInformationUtils.java:27)
    at io.prestosql.plugin.hive.authentication.DirectHdfsAuthentication.doAs(DirectHdfsAuthentication.java:38)
    at io.prestosql.plugin.hive.HdfsEnvironment.getFileSystem(HdfsEnvironment.java:70)
    at io.prestosql.plugin.hive.HdfsEnvironment.getFileSystem(HdfsEnvironment.java:64)
    at io.prestosql.plugin.hive.SyncPartitionMetadataProcedure.syncPartitionMetadata(SyncPartitionMetadataProcedure.java:114)
    ... 8 more
```

Also, once we got the exception, subsequent queries cause same exception.
```
presto:default> call system.sync_partition_metadata('default', 'page_views', 'add');
Query 20190308_133052_00000_ujnt9 failed: No FileSystem for scheme: hdfs

presto:default> select * from hive.default.t1;
Query 20190308_133252_00001_ujnt9 failed: No FileSystem for scheme: hdfs
```

If we execute any query to hive before calling sync_partition_metadata, this issue will not cause. 
```sql
presto:default> select * from hive.default.t1;
     c1     
------------
 0600-01-01 
(1 row)

presto:default> call system.sync_partition_metadata('default', 'page_views', 'add');
CALL
```

--- a/presto-hive/src/main/java/io/prestosql/plugin/hive/CreateEmptyPartitionProcedure.java
+++ b/presto-hive/src/main/java/io/prestosql/plugin/hive/CreateEmptyPartitionProcedure.java
@@ -21,6 +21,7 @@ import io.prestosql.plugin.hive.LocationService.WriteInfo;
 import io.prestosql.plugin.hive.PartitionUpdate.UpdateMode;
 import io.prestosql.plugin.hive.metastore.ExtendedHiveMetastore;
 import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.classloader.ThreadContextClassLoader;
 import io.prestosql.spi.connector.ConnectorSession;
 import io.prestosql.spi.procedure.Procedure;
 import io.prestosql.spi.procedure.Procedure.Argument;
@@ -83,6 +84,13 @@ public class CreateEmptyPartitionProcedure
     }
 
     public void createEmptyPartition(ConnectorSession session, String schema, String table, List<Object> partitionColumnNames, List<Object> partitionValues)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doCreateEmptyPartition(session, schema, table, partitionColumnNames, partitionValues);
+        }
+    }
+
+    private void doCreateEmptyPartition(ConnectorSession session, String schema, String table, List<Object> partitionColumnNames, List<Object> partitionValues)
     {
         TransactionalMetadata hiveMetadata = hiveMetadataFactory.get();
 

--- a/presto-hive/src/main/java/io/prestosql/plugin/hive/SyncPartitionMetadataProcedure.java
+++ b/presto-hive/src/main/java/io/prestosql/plugin/hive/SyncPartitionMetadataProcedure.java
@@ -21,6 +21,7 @@ import io.prestosql.plugin.hive.metastore.Partition;
 import io.prestosql.plugin.hive.metastore.SemiTransactionalHiveMetastore;
 import io.prestosql.plugin.hive.metastore.Table;
 import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.classloader.ThreadContextClassLoader;
 import io.prestosql.spi.connector.ConnectorSession;
 import io.prestosql.spi.connector.SchemaTableName;
 import io.prestosql.spi.connector.TableNotFoundException;
@@ -94,6 +95,13 @@ public class SyncPartitionMetadataProcedure
     }
 
     public void syncPartitionMetadata(ConnectorSession session, String schemaName, String tableName, String mode)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doSyncPartitionMetadata(session, schemaName, tableName, mode);
+        }
+    }
+
+    private void doSyncPartitionMetadata(ConnectorSession session, String schemaName, String tableName, String mode)
     {
         SyncMode syncMode = toSyncMode(mode);
         HdfsContext context = new HdfsContext(session, schemaName, tableName);

